libraries/AC_Avoidance/AP_OADatabase.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/Location.h>
+#include <AP_Param/AP_Param.h>
+#include <AP_HAL/utility/RingBuffer.h>
+
+class AP_OADatabase {
+public:
+
+    AP_OADatabase();
+
+    /* Do not allow copies */
+    AP_OADatabase(const AP_OADatabase &other) = delete;
+    AP_OADatabase &operator=(const AP_OADatabase&) = delete;
+
+    struct OA_DbItem {
+        Location loc;
```

Comment:
For what you need Location is a very expensive object to be storing in the queue. All you need is 8 bytes for lat/lng, and instead you are getting all the flags + altitude coming along with it. For what you are doing it's probably better to store this as 2 int32_t's and make a distance helper that can do math directly on that. (Or alternatively Vector2l). This would cut the cost of these structs from 20 bytes per, to 12 bytes per. This is a considerable size difference, which makes this a lot easier to fit on the embedded boards, and on anything that has a cache (the M7's and up) a lot faster.